aws/resource_aws_redshift_cluster.go

```diff
@@ -244,6 +244,9 @@ func resourceAwsRedshiftCluster() *schema.Resource {
 				Type:     schema.TypeList,
 				MaxItems: 1,
 				Optional: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return old == "1" && new == "0"
```

Comment:
Not sure exactly what this does? Does this suppress a diff if you remove the last one?

Reply:
This suppresses when:
* The resource schema includes an optional configuration block with defaults
* The API response includes those defaults to refresh into the Terraform state
* The operator's configuration omits the optional configuration block

In the above scenario, no differences are reported by Terraform except for the configuration block attribute itself `my_config_block_attr.#: "1" => "0"`. This little `DiffSuppressFunc` fixes that.

I feel like there is an unmerged pull request somewhere that actually creates this in a small reusable function with an appropriate name of `DiffSuppressMissingOptionalConfigurationBlock` (or I'd name it something like that anyways). Ideally, we should replace all the various implementations of this suppression with that function for clarity. 😅